sdk/src/trace/span.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "src/trace/tracer.h"
+
+#include <mutex>
+
+namespace opentelemetry
+{
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+
+class Span final : public trace_api::Span
+{
+public:
+  explicit Span(std::shared_ptr<Tracer> &&tracer,
+                nostd::string_view name,
+                const trace_api::StartSpanOptions &options) noexcept;
+
+  ~Span() override;
+
+  void AddEvent(nostd::string_view name) noexcept override;
+
+  void AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) noexcept override;
```

Comment:
I was hoping to discuss this in the PR that added the Span API, but can we have just one variant of these?

Reply:
I dropped the AddEvent with the steady timestamp